### PR TITLE
fix error message to make it clear and precise

### PR DESCRIPTION
**Pull request by Joeper214**
PR is for 

https://skyarch.backlog.jp/view/NC-526

### CODE REVIEW

**Comment by Joeper214** [comment]
I think the test coverage is broken again. @pocke

File: app/controllers/monitorings_controller.rb [hunk 1]
```diff
@@ -112,7 +112,15 @@ def show_zabbix_graph
       item_key = item_infos.first["key_"]
     end
 
-    history_all = z.get_history(physical_id, item_key)
+
+    begin
+      history_all = z.get_history(physical_id, item_key)
+      rescue SyntaxError, NameError => boom
```
Comment (skyarch-networks) [inline app/controllers/monitorings_controller.rb]: __[rubocop]__
* Style/RescueEnsureAlignment: `rescue` at 118, 6 is not aligned with `end` at 122, 4.
* Lint/UselessAssignment: Useless assignment to variable - `boom`.

File: app/controllers/monitorings_controller.rb [hunk 2]
```diff
@@ -112,7 +112,15 @@ def show_zabbix_graph
...
+        raise item_key.to_s+ ": was not selected in zabbix Templates. Please select template in the templates tab."
+      rescue StandardError => bang
```
Comment (skyarch-networks) [inline app/controllers/monitorings_controller.rb]: __[rubocop]__
* Style/RescueEnsureAlignment: `rescue` at 120, 6 is not aligned with `end` at 122, 4.

File: app/controllers/monitorings_controller.rb [hunk 3]
```diff
@@ -112,7 +112,15 @@ def show_zabbix_graph
...
+        puts "Error running script: " + bang.to_s
```
Comment (skyarch-networks) [inline app/controllers/monitorings_controller.rb]: __[rubocop]__
* Rails/Output: Do not write to stdout. Use Rails' logger if you want to log.

File: app/controllers/monitorings_controller.rb [hunk 4]
```diff
@@ -112,7 +112,13 @@ def show_zabbix_graph
       item_key = item_infos.first["key_"]
     end
 
-    history_all = z.get_history(physical_id, item_key)
+
+    begin
+      history_all = z.get_history(physical_id, item_key)
+      rescue SyntaxError, NameError => boom
```
Comment (skyarch-networks) [inline app/controllers/monitorings_controller.rb]: __[rubocop]__
* Style/RescueEnsureAlignment: `rescue` at 118, 6 is not aligned with `end` at 120, 4.
* Lint/UselessAssignment: Useless assignment to variable - `boom`.

File: app/controllers/monitorings_controller.rb [hunk 5]
```diff
@@ -112,7 +112,13 @@ def show_zabbix_graph
       item_key = item_infos.first["key_"]
     end
 
-    history_all = z.get_history(physical_id, item_key)
+
+    begin
+      history_all = z.get_history(physical_id, item_key)
+      rescue SyntaxError, NameError
```
Comment (skyarch-networks) [inline app/controllers/monitorings_controller.rb]: __[rubocop]__
* Style/RescueEnsureAlignment: `rescue` at 118, 6 is not aligned with `end` at 120, 4.

File: app/controllers/monitorings_controller.rb [hunk 6]
```diff
@@ -112,7 +112,13 @@ def show_zabbix_graph
       item_key = item_infos.first["key_"]
     end
 
-    history_all = z.get_history(physical_id, item_key)
+
+    begin
+      history_all = z.get_history(physical_id, item_key)
+    rescue SyntaxError, NameError
+        raise item_key.to_s + I18n.t('monitoring.msg.not_set')
```
Comment (skyarch-networks) [inline app/controllers/monitorings_controller.rb]: __[rubocop]__
* Style/IndentationWidth: Use 2 (not 4) spaces for indentation.

**Comment by pocke** [comment]
Now, you catch `NoMethodError` at monitorings_controller(`NoMethodError` extends `NameError`).
It's one of solution, but maybe this solution catch other exceptions.

Cause of `NoMethodError` is result of `get_item_info`(at `app/model/zabbix.rb` Line403) to be empty.
I think we should check result of `get_item_info` isn't empty. And raise error in `get_history()` function.

**Comment by Joeper214** [comment]
:+1:

File: app/models/zabbix.rb [hunk 7]
```diff
@@ -402,6 +402,8 @@ def get_user_type_by_user(user)
   def get_history(physical_id, item_key)
     item_info = get_item_info(physical_id, item_key, "filter")
 
+    puts "=====info====="
```
Comment (skyarch-networks) [inline app/models/zabbix.rb]: __[rubocop]__
* Rails/Output: Do not write to stdout. Use Rails' logger if you want to log.

File: app/models/zabbix.rb [hunk 8]
```diff
@@ -402,6 +402,8 @@ def get_user_type_by_user(user)
...
+    puts item_info.inspect
```
Comment (skyarch-networks) [inline app/models/zabbix.rb]: __[rubocop]__
* Rails/Output: Do not write to stdout. Use Rails' logger if you want to log.

File: app/models/zabbix.rb [hunk 9]
```diff
@@ -412,15 +412,19 @@ def get_history(physical_id, item_key)
...
-      history: type,
-      itemids: item_info.first["itemid"],
-      sortfield: 'clock',
-      sortorder: 'DESC',
-      limit: 30,
-    )
-
+      case item_info
+      when []
+        raise item_key.to_s + I18n.t('monitoring.msg.not_set')
```
Comment (pocke) [inline app/models/zabbix.rb]:
```ruby
item_info = get_item_info(...)
raise <Exception Class>, <Message> if item_info.blank?
...
```

### `blank?` method

### Not `RuntimeError`

default

We get error message as a `RuntimeError`. 
However, we can specify kind of Error.

Please define Exception class, and raise this(e.g. https://github.com/skyarch-networks/skyhopper/blob/master/app/models/node.rb#L15-L17).

Comment (Joeper214) [inline app/models/zabbix.rb]: I see. Thanks.

**Comment by pocke** [comment]
I'll merge at next week.

**Comment by Joeper214** [comment]
Okay. Thanks!

**Comment by pocke** [comment]
Thanks :+1:

**Pull request merged**